API: add method to set v: variable

pqzx opened the pull request:
#9349

pqzx committed:
API: nvim_set_vvar(): set v: variables #9395

closes #9349

justinmk commented:
Thanks. Next step would be:

- check the failing CI logs. They will tell you that `FUNC_API_SINCE(1)` is wrong, because the current API version is not 1, and this function didn't exist at API version 1.
- the new feature needs test coverage. see `test/functional/api` for examples.

justinmk commented:
Looks like this wasn't tested. It does not work.

justinmk committed:
dict_set_var: check value before checking its container

- When setting a fixed/locked/readonly var, it is more relevant to
  report on the key, not its container dict. If its container dict (v:)
  is readonly, that does not mean the key itself is readonly.
- Allow modifying a "fixed" var. "fixed" only prevents deletion.

justinmk committed:
test/API: nvim_set_vvar() #9395

justinmk commented on src/nvim/api/private/helpers.c:
```diff
@@ -184,35 +184,28 @@ Object dict_set_var(dict_T *dict, String key, Object value, bool del,
       api_set_error(err, kErrorTypeException, "Key is read-only: %s", key.data);
       return rv;
-    } else if (di->di_flags & DI_FLAGS_FIX) {
-      api_set_error(err, kErrorTypeException, "Key is fixed: %s", key.data);
```
IIUC the docstring at `DI_FLAGS_FIX` implies that a "fixed" value may be modified but not deleted. So I changed it to `(del && (di->di_flags & DI_FLAGS_FIX))` .
